lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -1179,23 +1179,39 @@ sub job_schedule_iso {
         asset_register(%$a);
     }
     my $noobsolete = delete $args{_NOOBSOLETEBUILD};
-    # ISOURL == ISO download. If the ISO already exists, skip the
-    # download step (below) entirely by leaving $isodlpath unset.
-    my $isodlpath;
-    if ($args{ISOURL}) {
+    # Any arg name ending in _URL is special: it tells us to download
+    # the file at that URL before running the job
+    my %downloads = ();
+    for my $arg (keys %args) {
+        next unless ($arg =~ /_URL$/);
```

Comment:
I'm sorry, I'm not sure I quite get the question. Are you saying you happen to have an existing parameter which ends in _URL which isn't for downloading?

Hmm - I guess what we can do is reverse the ordering just below this chunk, so we check if its a downloadable type before we generate the shortened param, and complain less when it's not downloadable? Sound good?